test/test_cache.py

```diff
@@ -326,6 +326,35 @@ def test_package_json_missing_attributes(self):
         for attr in ['name', 'version', 'release', 'arch', 'vendor']:
             self.assertEqual(None, data[attr])
 
+    def test_module_md_uniquify(self):
+        modules_input = [
+            {
+                "name": "duck",
+                "stream": 0,
+                "version": "20180730233102",
+                "context": "deadbeef",
+                "arch": "noarch",
+                "profiles": ["default"],
+                "installed_profiles": [],
+                "status": "enabled"
+            },
+            {
+                "name": "duck",
+                "stream": 0,
+                "version": "20180707144203",
+                "context": "c0ffee42",
+                "arch": "noarch",
+                "profiles": ["default", "server"],
+                "installed_profiles": ["server"],
+                "status": "unknown"
+            }
+
+        ]
+
+        self.assertEqual(modules_input, ModulesProfile.__uniquify(modules_input))
```

Comment:
It is not allowed to call "private" methods in this way in Python. When you create private method like this:

```python
class C(object):
    __private(self):
        print("Private method")
```

Then you can not do something like this:

```python
c = C()
c.__private()
```

But you can use this trick:

```python
c = C()
c._C__private()
```